modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -340,6 +341,18 @@ public BlobContainer blobContainer(BlobPath path) {
         return new S3BlobContainer(path, this);
     }
 
+    private static class DeletionExceptions {
+        private Exception exception = null;
```

Comment:
May as well be package-private since we access it from outside this class.

```suggestion
        Exception exception = null;
```